qhub/template/{{ cookiecutter.repo_directory }}/infrastructure/modules/kubernetes/services/prefect/chart/templates/prefect.yaml

```diff
@@ -23,6 +23,10 @@ spec:
             - /bin/bash
             - -c
           env:
+            {{- range $k, $v := .Values.envVars }}
+            - name: {{ $k }}
+              value: {{ $v -}}
+            {{ end }}
```

Comment:
@viniciusdc I added these lines.  They allow you to add arbitrary environment variables to the prefect agent.  This was one thing that was requested by jhamman, and may simplify this PR.  This also allows the users to specify any environment variable including others that could affect prefect without requiring us to make a PR to qhub in the future.  

E.g. A values file like the one below . . . 
```yaml
# values.yaml
prefectImage: <placeholder>
namespace: <placeholder>
serviceAccount: prefect
prefectToken: "<placeholder>"
cloudApi: "<placeholder>"
env: 
  MY_ENV_VAR: my_env_var_value
  MY_ENV_VAR2: my_env_var_value2
  MY_ENV_VAR3: my_env_var_value3
```
. . . would produce the following kubernetes manifest.
```
# Source: prefect/templates/prefect.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: prefect-agent
  name: prefect-agent
spec:
  replicas: 1
  selector:
    matchLabels:
      app: prefect-agent
  template:
    metadata:
      labels:
        app: prefect-agent
    spec:
      serviceAccountName: prefect
      containers:
        - args:
            - prefect agent start kubernetes -e JUPYTERHUB_API_TOKEN=$JUPYTERHUB_API_TOKEN
          command:
            - /bin/bash
            - -c
          env:
            - name: MY_ENV_VAR
              value: my_env_var_value
            - name: MY_ENV_VAR2
              value: my_env_var_value2
            - name: MY_ENV_VAR3
              value: my_env_var_value3
# . . . rest of file would be below
```

Reply:
Oh that's nice! thanks for the pointer Adam

Reply:
My only concern with that is related with the variables that need use `toJson` when included in the manifest, e.g. `PREFECT__CLOUD__AGENT__LABELS` which is a list and `IMAGE_PULL_SECRETS` which is a dictionary containing the secrets. How would I handle that using the proposed env structure above?

Reply:
I imagine you could get the desired format by using back slashes.  If not, I think it's reasonable to still keep the environment variables that need special formatting in there.